system-addon/lib/ManualMigration.jsm

```diff
@@ -23,20 +25,22 @@ this.ManualMigration = class ManualMigration {
     Services.obs.removeObserver(this, MIGRATION_ENDED_EVENT);
   }
 
-  isMigrationMessageExpired() {
+  async isMigrationMessageExpired() {
     let migrationLastShownDate = new Date(this._prefs.get("migrationLastShownDate") * 1000);
     let today = new Date();
     // Round down to midnight.
     today = new Date(today.getFullYear(), today.getMonth(), today.getDate());
-
     if (migrationLastShownDate < today) {
+      let profileAge = new ProfileAge(null, null);
+      let profileCreationDate = await profileAge.created;
+      let daysSinceProfileCreation = (Date.now() - profileCreationDate) / MS_PER_DAY;
```

Comment:
Any reason to do this only inside ` if (migrationLastShownDate < today)` ? It seems like all the profile age stuff can be done at the beginning of `isMigrationMessageExpired` and just short circuit the rest of the method? It would also avoid setting the remaining/lastshown prefs below… unless that's desired to set?